internal/controllers/machine/machine_controller_phases.go

```diff
@@ -116,9 +116,8 @@ func (r *Reconciler) reconcileExternal(ctx context.Context, cluster *clusterv1.C
 		return external.ReconcileOutput{}, err
 	}
 
```

Comment:
Is the general idea:

* on Machine creation we first create the BootstrapConfig / InfraMachine then the Machine
* BootstrapConfig / InfraMachine are created with ownerRef to KCP / MS in case the Machine creation fails so that they get eventually cleaned up on KCP / MS deletion
* During Machine reconcile we can remove the KCP / MS ownerRef as we then don't need them anymore